File: fyc-api-rest/src/main/java/fr/esgi/fyc_api_rest/service/SeanceService.java
package fr.esgi.fyc_api_rest.service;

import fr.esgi.fyc_api_rest.business.Seance;

import java.util.List;

public interface SeanceService {
    Seance create(Seance seance);
    List<Seance> findAll();
    Seance findById(Long id);

    void delete(Long id);

    Seance update(Long id, Seance entity);
}

File: fyc-rest-api-csharp-test/CometeAPI/Service/SeanceService.cs
using FycRestApi.Business;
using FycRestApi.Repository;
using FycRestApi.Service.Interface;

namespace FycRestApi.Service;

public class SeanceService : ISeanceService
{
    private readonly SeanceRepository _repo;
    public SeanceService(SeanceRepository seanceRepository)
    {
        _repo = seanceRepository;
    }

    public async Task delete(int id)
    {
        await _repo.delete(id);
    }

    public Task<List<Seance>> findAll()
    {
        return _repo.findAll();
    }

    public Task<Seance> findById(int id)
    {
        return _repo.findById(id);
    }

    public async Task<Seance> save(Seance seance)
    {
        return await _repo.save(seance);
    }

    public Task<Seance> update(int id, Seance seance)
    {
        return _repo.update(id, seance);
    }
}